emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBSerDe.java

```diff
@@ -207,8 +208,9 @@ private void verifyDynamoDBWriteThroughput(Configuration conf, Properties tbl) {
       throw new RuntimeException("Could not get cluster capacity.", e);
     }
 
+    BillingModeSummary billingModeSummary = client.describeTable(dynamoDBTableName).getBillingModeSummary();
     if (maxMapTasks > writesPerSecond &&
-            client.describeTable(dynamoDBTableName).getBillingModeSummary().getBillingMode().equals(DynamoDBConstants.BILLING_MODE_PROVISIONED)) {
+            billingModeSummary == null || billingModeSummary.getBillingMode().equals(DynamoDBConstants.BILLING_MODE_PROVISIONED)) {
```

Comment:
this line need to be adjusted with adding `( )` to avoid logic error 

`if (maxMapTasks > writesPerSecond && 
(billingModeSummary == null || billingModeSummary.getBillingMode().equals(DynamoDBConstants.BILLING_MODE_PROVISIONED)))`

Reply:
DOH! Fixed. Thanks.